ResearchKit/Charts/ORKGraphChartView.m

```diff
@@ -713,53 +774,50 @@ - (CGFloat)snappedXPosition:(CGFloat)xPosition {
     return xPosition;
 }
 
-- (CGFloat)valueForCanvasXPosition:(CGFloat)xPosition {
+- (CGFloat)valueForCanvasXPosition:(CGFloat)xPosition plotIndex:(NSInteger)plotIndex {
     BOOL snapped = [self isXPositionSnapped:xPosition];
     CGFloat value = ORKCGFloatInvalidValue;
-    NSUInteger positionIndex = 0;
     if (snapped) {
-        CGFloat numberOfXAxisPoints = self.numberOfXAxisPoints;
+        NSInteger positionIndex = 0;
+        NSInteger numberOfXAxisPoints = self.numberOfXAxisPoints;
         for (positionIndex = 0; positionIndex < (numberOfXAxisPoints - 1); positionIndex++) {
             CGFloat xAxisPointValue = xAxisPoint(positionIndex, numberOfXAxisPoints, _plotView.bounds.size.width);
             if (xAxisPointValue == xPosition) {
                 break;
             }
         }
-        value = ((ORKRangedPoint *)_dataPoints[0][positionIndex]).maximumValue;
+        value = ((ORKRangedPoint *)_dataPoints[plotIndex][positionIndex]).maximumValue;
```

Comment:
Would it be better to create a helper method to retrieve `ORKRangedPoint` from the nested array? So we don't need to cast it each time.